registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java

```diff
@@ -261,49 +269,52 @@ public MessageDTO process(MessageDTO object) {
 							uinField, description);
 
 					boolean isUinAlreadyPresent = isUinAlreadyPresent(idResponseDTO, registrationId);
+					
+					if (isUinAlreadyPresent) {
 
-					if (isIdResponseNotNull(idResponseDTO) || isUinAlreadyPresent) {
-						generateVid(registrationId, uinResponseDto.getResponse().getUin(), isUinAlreadyPresent);
-						registrationStatusDto.setStatusComment(StatusUtil.UIN_GENERATED_SUCCESS.getMessage());
-						registrationStatusDto.setSubStatusCode(StatusUtil.UIN_GENERATED_SUCCESS.getCode());
-						String uinStatus = isUinAlreadyPresent ? UINConstants.UIN_UNASSIGNED : UINConstants.UIN_ASSIGNED;
-						sendResponseToUinGenerator(registrationId, uinResponseDto.getResponse().getUin(),
```

Comment:
In case of uin already present, do we get the uin sent in request back in response ?

Reply:
fixed